react-basic-practice/src/components/ProductCard/index.jsx

```diff
@@ -1,18 +1,31 @@
 import React from 'react';
+import { Link } from 'react-router-dom';
 import './index.css';
 
 const ProductCard = (props) => {
   const { product } = props;
-  const price = product.price.toFixed(2);
+  let label = '';
+
+  const showLabel = () => {
+    let haveLabel = false;
+    if (!product.quantity) {
+      label = 'Sold out';
```

Comment:
you should not have side effect here. This needs to be a pure function. Please google it.